app/js/components/service_form.test.js

```diff
@@ -0,0 +1,39 @@
+jest
+    .dontMock('fs')
+    .dontMock('jquery');
+
+
+window.$ = require('jquery');
+
+let html = require('fs').readFileSync('./app/js/components/mock/serviceForm.html').toString();
```

Comment:
Cool solution for integration testing. My only concern with this kind of test is that they are very fragile. If the form changes in the slightest, the test becomes unpredictable/unstable. 

Some thoughts, and feel free to discuss them IRL

* Can we further abstract the DOM from these tests? In a way that only mockable elements are used in the javascript and we only have to provide mock objects for these elements?
* Do we need to update our test plan? 
* Do we need to update our DoD with a trigger to update test files when certain changes have been applied?

Reply:
I think for now this solution will be sufficient. In order to have a more predictable/stable outcome we should consider js end-to-end testing but for now in my opinion that's too much work. I agree that it's better to use a more component based approach with rigorous  unit testing for the different components. But I think for now we're good off as is. Because it will be very easy to copy-paste the form from a browser.

I think if you're testing something you always have a commitment in keeping the mocks/fixtures/tests etc. updated. Should you then define that in your DOD or testplan? I don't mind however if we put it there. What's your opinion?

Reply:
Yes, the test shouldn't be too hard to maintain. I'm fine with this!